apis/v1alpha1/gateway_types.go

```diff
@@ -108,8 +108,10 @@ type GatewaySpec struct {
 	Addresses []GatewayAddress `json:"addresses"`
 }
 
-// Listener embodies the concept of a logical endpoint where a
-// Gateway can accept network connections.
+// Listener embodies the concept of a logical endpoint where a Gateway can
+// accept network connections. Each listener in a Gateway must have a unique
+// combination of Hostname, Port, and Protocol. This will be enforced by a
+// validating webhook.
```

Comment:
@robscott @bowei will this validating webhook be part of the service-apis repo?

Reply:
Yep, that's the plan. There are a number of validations that simply can't be accomplished with kubebuilder annotations at this point. I believe this would be OK post alpha API launch though.